From hidave.darkstar@gmail.com  Tue May 13 15:00:13 2008
From: Dave Young <hidave.darkstar@gmail.com>
Date: Fri, 9 May 2008 15:24:08 +0800
Subject: Driver core: struct class remove children list
To: gregkh@suse.de
Cc: linux-kernel@vger.kernel.org
Message-ID: <20080509072408.GC2740@darkstar.te-china.tietoenator.com>
Content-Disposition: inline


because of the class_device was removed, now do the children list removing

Signed-off-by: Dave Young <hidave.darkstar@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/base/class.c   |    1 -
 include/linux/device.h |    1 -
 2 files changed, 2 deletions(-)

--- a/drivers/base/class.c
+++ b/drivers/base/class.c
@@ -140,7 +140,6 @@ int class_register(struct class *cls)
 
 	pr_debug("device class '%s': registering\n", cls->name);
 
-	INIT_LIST_HEAD(&cls->children);
 	INIT_LIST_HEAD(&cls->devices);
 	INIT_LIST_HEAD(&cls->interfaces);
 	kset_init(&cls->class_dirs);
--- a/include/linux/device.h
+++ b/include/linux/device.h
@@ -183,7 +183,6 @@ struct class {
 	struct module		*owner;
 
 	struct kset		subsys;
-	struct list_head	children;
 	struct list_head	devices;
 	struct list_head	interfaces;
 	struct kset		class_dirs;
